appserver/web/web-glue/src/main/java/com/sun/enterprise/web/WebModuleListener.java

```diff
@@ -189,12 +189,17 @@ private void configureJsp(WebModule webModule) {
         servletContext.setAttribute("com.sun.appserv.tldlistener.map", tldListenerMap);
 
         ServiceLocator defaultServices = webContainer.getServerContext().getDefaultServices();
-        final String servicesName = webModule.getComponentId();
-        ServiceLocator webAppServices = ServiceLocatorFactory.getInstance().create(servicesName, defaultServices);
-        initializeServicesFromClassLoader(webAppServices, Thread.currentThread().getContextClassLoader());
 
-        // set services for jsf injection
-        servletContext.setAttribute(Constants.HABITAT_ATTRIBUTE, webAppServices);
+        if (webModule.isSystemApplication()) {
+            final String servicesName = webModule.getComponentId();
+            ServiceLocator webAppServices = ServiceLocatorFactory.getInstance().create(servicesName, defaultServices);
```

Comment:
Can tou move creation of webAppServices to a separate method for clarity?

Reply:
I mean, refactor the code, so that the code to create webAppServices is extracted into a new method.